### PR TITLE
Feature/am 271

### PR DESCRIPTION
**Pull request by abhijitdiwan**
### Checklist ###

- [ ] PR title follows [conventional commits](https://www.conventionalcommits.org) approach and uses one of the following types:
    - build: changes that affect the build system or external dependencies (example scopes: gulp, broccoli, npm)
    - ci: changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
    - docs: documentation only changes
    - feat: a new feature
    - fix: a bug fix
    - perf: a code change that improves performance
    - refactor: a code change that neither fixes a bug nor adds a feature
    - style: changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
    - test: adding missing tests or correcting existing tests
- [ ] commit messages are meaningful and follow good commit message guidelines to ease review process
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

### CODE REVIEW

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java [hunk 1]
```diff
@@ -106,6 +110,47 @@ void whenInsufficientRolePermissionsShouldReturnNull() {
         assertThat(rolePermissions).isNull();
     }
 
+    @Test
+    void whenRolePermissionIsHighestThenShowAllAttribute() {
```
Comment (geek-soft) [inline am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java]: @abhijitdiwan on the basis that the tests does not exercise any new execution path it adds no much more value. 

However as see value in it as it shows that not only exact match on security classification is possible but all security classifications at the same or below level of hierarchy are fetched. On the above basis I suggest changing existing `whenGettingRolePermissionsShouldReturnPermissionsAndSecurityClassificationsForSpecifiedRole` by replicating the scenario that you proposed.

In other words no new test please just change to existing one.

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java [hunk 2]
```diff
@@ -106,6 +110,47 @@ void whenInsufficientRolePermissionsShouldReturnNull() {
...
+            service.getRolePermissions(buildResource(serviceName, RESOURCE_TYPE, RESOURCE_NAME), roleName);
+
+
+        assertThat(rolePermissions.getPermissions())
+            .hasSize(3)
+            .contains(new AbstractMap.SimpleEntry<>(JsonPointer.valueOf("/address"), ImmutableSet.of(READ)))
+            .containsEntry(JsonPointer.valueOf("/address/line1"), multiPermission)
+            .contains(new AbstractMap.SimpleEntry<>(JsonPointer.valueOf(""),ImmutableSet.of(READ)));
+    }
+
```
Comment (swalker125) [inline am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java]: A lot of this is out of date with the most recent commit to AM-271-security-classifications-for-resource-attribute

**Comment by swalker125** [comment]
I would also consider renaming the PR title using https://www.conventionalcommits.org/en/v1.0.0-beta.2/ (this isn't a feature) it is also unclear what the PR does from the title.

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java (hunk 1 above)
Comment (abhijitdiwan) [reply am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java]: Updated in existing method. Please have a look

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java [hunk 3]
```diff
@@ -42,13 +45,23 @@ void setUp() {
 
     @Test
     void whenGettingRolePermissionsShouldReturnPermissionsAndSecurityClassificationsForSpecifiedRole() {
+
+        ImmutableSet multiPermission = ImmutableSet.builder().add(READ).add(CREATE).add(UPDATE).build();
```
Comment (swalker125) [inline am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java]:
```suggestion
        Set<Permission> multiPermission = ImmutableSet.of(READ, CREATE,UPDATE);
```

Comment (swalker125) [reply am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java]: although personally I think this is uneccessary complication for the test. The logic does nothing with permissions, so tbh I would actually remove

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java [hunk 4]
```diff
@@ -42,13 +45,23 @@ void setUp() {
...
+
         Map.Entry<Set<Permission>, SecurityClassification> publicReadPermission =
             new Pair<>(ImmutableSet.of(READ), PUBLIC);
 
-        addRoleWithSecurityClassification(roleName, PUBLIC);
+        Map.Entry<Set<Permission>, SecurityClassification> privateReadPermission =
+            new Pair<>(ImmutableSet.of(READ), PRIVATE);
+
+        Map.Entry<Set<Permission>, SecurityClassification> restrictedReadPermission =
```
Comment (swalker125) [inline am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java]: if you keep your multiPermissions this is no longer a restrictedReadPermission

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java [hunk 5]
```diff
@@ -106,6 +121,7 @@ void whenInsufficientRolePermissionsShouldReturnNull() {
         assertThat(rolePermissions).isNull();
     }
 
+
```
Comment (swalker125) [inline am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java]: unneccesary whitespace has been added here.

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java (hunk 4 above)
Comment (abhijitdiwan) [reply am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java]: yes right

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java [hunk 6]
```diff
@@ -42,13 +46,23 @@ void setUp() {
 
     @Test
     void whenGettingRolePermissionsShouldReturnPermissionsAndSecurityClassificationsForSpecifiedRole() {
+
+        ImmutableSet multiPermission = ImmutableSet.builder().add(READ).add(CREATE).add(UPDATE).add(DELETE).build();
```
Comment (geek-soft) [inline am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java]: Change immutable set to interface to stick to the rule advising coding against interfaces.

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java (hunk 5 above)
Comment (abhijitdiwan) [reply am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java]: replaced

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java (hunk 4 above)
Comment (abhijitdiwan) [reply am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java]: done

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java (hunk 6 above)
Comment (geek-soft) [reply am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java]: Why is that extracted to variable if other sets are not? Be consistent, I do not think you need that variable really.

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java [hunk 7]
```diff
@@ -42,13 +46,23 @@ void setUp() {
 
     @Test
     void whenGettingRolePermissionsShouldReturnPermissionsAndSecurityClassificationsForSpecifiedRole() {
+
```
Comment (geek-soft) [inline am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java]: Unnecessary blank line, please remove.

**Comment by swalker125** [comment]
I would refrain from copy and pasting commit messages. I know we're going to squash, but I have no idea what the difference between commits is in this PR.

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java (hunk 6 above)
Comment (abhijitdiwan) [reply am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java]: done

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java (hunk 7 above)
Comment (abhijitdiwan) [reply am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java]: done

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java [hunk 8]
```diff
@@ -45,10 +49,18 @@ void whenGettingRolePermissionsShouldReturnPermissionsAndSecurityClassifications
         Map.Entry<Set<Permission>, SecurityClassification> publicReadPermission =
             new Pair<>(ImmutableSet.of(READ), PUBLIC);
 
-        addRoleWithSecurityClassification(roleName, PUBLIC);
+        Map.Entry<Set<Permission>, SecurityClassification> privateReadPermission =
+            new Pair<>(ImmutableSet.of(READ), PRIVATE);
+
+        Map.Entry<Set<Permission>, SecurityClassification> restrictedAllPermission =
+            new Pair<>(ImmutableSet.of(CREATE,READ,UPDATE,DELETE), RESTRICTED);
```
Comment (swalker125) [inline am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java]:
```suggestion
            new Pair<>(ImmutableSet.of(CREATE, READ, UPDATE, DELETE), RESTRICTED);
```

Comment (abhijitdiwan) [reply am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java]: done

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java [hunk 9]
```diff
@@ -57,14 +69,16 @@ void whenGettingRolePermissionsShouldReturnPermissionsAndSecurityClassifications
         assertThat(rolePermissions).isEqualTo(RolePermissions.builder()
             .permissions(ImmutableMap.of(
                 JsonPointer.valueOf("/child"), ImmutableSet.of(READ),
-                JsonPointer.valueOf("/parent/age"), ImmutableSet.of(READ)
+                JsonPointer.valueOf("/parent/age"), ImmutableSet.of(READ),
+                JsonPointer.valueOf("/parent/address"), ImmutableSet.of(CREATE,READ,UPDATE,DELETE)
```
[the remark this answers is not in the document]
Comment (abhijitdiwan) [reply am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java]: done

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java [hunk 10]
```diff
@@ -45,10 +49,18 @@ void whenGettingRolePermissionsShouldReturnPermissionsAndSecurityClassifications
...
 
-        addRoleWithSecurityClassification(roleName, PUBLIC);
+        Map.Entry<Set<Permission>, SecurityClassification> privateReadPermission =
+            new Pair<>(ImmutableSet.of(READ), PRIVATE);
+
+        Map.Entry<Set<Permission>, SecurityClassification> restrictedAllPermission =
+            new Pair<>(ImmutableSet.of(CREATE, READ, UPDATE, DELETE), RESTRICTED);
+
+        addRoleWithSecurityClassification(roleName, RESTRICTED);
+
```
[the remark this answers is not in the document]
Comment (abhijitdiwan) [reply am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java]: done

**Comment by abhijitdiwan** [comment]
Will create new branch with conventional commit
